corehq/apps/translations/app_translations/utils.py

```diff
@@ -202,6 +205,49 @@ def none_or_unicode(val):
     return rows
 
 
+def text_and_audio_changed(current_row, previous_row, headers):
+    text_changed = False
+    audio_path_changed = False
+    for index, header in enumerate(headers):
+        if "default_" in header and previous_row[index] != current_row[header]:
+            text_changed = True
+        if "audio_" in header and previous_row[index] != current_row[header]:
+            audio_path_changed = True
+    return text_changed, audio_path_changed
+
+
+def generate_audio_path(text, lang, previous_path):
+    path_hash = md5(f'{text}_{lang}'.encode()).hexdigest()[:6]
```

Comment:
is this the same implementation that is done in app builder/vellum as well?

Reply:
I checked the length of the random characters that are generated and they are six and the implementation on Vellum does not matter here because we are not generating any new audio path. If text is changed but there is no audio path configured before then we will not add an audio path on our own.
So this makes it independent of the implementation on Vellum.